addons/point_of_sale/static/src/js/models.js

```diff
@@ -2971,6 +3071,184 @@ export class Order extends PosModel {
+                    new: changes["new"],
+                    cancelled: changes["cancelled"],
+                    table_name: this.pos.config.is_table_management ? this.getTable().name : false,
+                    floor_name: this.pos.config.is_table_management
+                        ? this.getTable().floor.name
+                        : false,
+                    name: this.name || "unknown order",
+                    time: {
+                        hours,
+                        minutes,
+                    },
+                };
+                const receipt = QWeb.render("OrderChangeReceipt", { changes: printingChanges });
+                const result = await printer.print_receipt(receipt);
+                if (!result.successful) {
+                    isPrintSuccessful = false;
+                }
+            }
+        }
+
+        return isPrintSuccessful;
+    }
+    _getPrintingCategoriesChanges(categories) {
+        const currentOrderChange = this.changesToOrders;
+        return {
+            new: currentOrderChange["new"].filter((change) =>
+                this.pos.db.is_product_in_category(categories, change["product_id"])
+            ),
+            cancelled: currentOrderChange["cancelled"].filter((change) =>
+                this.pos.db.is_product_in_category(categories, change["product_id"])
+            ),
+        };
+    }
+    /**
+     * This function is called after the order has been successfully sent to the preparation tool(s).
+     * In the future, this status should be separated between the different preparation tools,
+     * so that if one of them returns an error, it is possible to send the information back to it
+     * without impacting the other tools.
+     */
+    updateLastOrderChange() {
+        this.orderlines.forEach((line) => {
+            if (!line.skipChange) {
+                const note = line.get_note();
+                const lineKey = `${line.uuid} - ${note}`;
+                if (this.lastOrderPrepaChange[lineKey]) {
+                    this.lastOrderPrepaChange[lineKey]["quantity"] = line.get_quantity();
+                } else {
+                    this.lastOrderPrepaChange[lineKey] = {
+                        line_uuid: line.uuid,
+                        product_id: line.get_product().id,
+                        name: line.get_full_product_name(),
+                        note: note,
+                        quantity: line.get_quantity(),
+                    };
+                }
+                line.setHasChange(false);
+            }
+        });
+    }
+
+    /**
+     * @returns {{ [productKey: string]: { product_id: number, name: string, note: string, quantity: number } }}
+     * This function recalculates the information to be sent to the preparation tools,
+     * it uses the variable lastOrderPrepaChange which contains the last changes sent
+     * to perform this calculation.
+     */
+    get computeOrderChanges() {
```

Comment:
Let's not write a getter if it performs a mutation. Getters that mutate the state is very confusing.

Rename to `getOrderChanges`. I know it's name is very similar, but with this, we know that we are calling a function when we try to "get the order changes".